### PR TITLE
feat(themes): add light_cycle theme — 24-hour radial clock

### PR DESCRIPTION
**Pull request by gkoch02**
A full-canvas circular clock that visualises the entire day at a glance:
twilight bands ring the rim (computed from astronomical / nautical / civil
twilight via src.astronomy when latitude/longitude are available, else
falling back to OWM-reported sunrise/sunset), today's calendar events
appear as radial dashes inside the ring, a needle marks the current
moment, and a sun (or moon, when night) glyph rides the rim at the
current-time position. Center disc carries day name, big date numeral,
month, and weather summary; footer reports rise/set/event count.

The theme registers with inky_palette=(INKY_YELLOW, INKY_BLUE) so the
sun + needle accent yellow and the night band picks up blue on Spectra-6
displays. On Waveshare 1-bit, twilight density is encoded by radial-dash
spacing (deep night = solid wedge, civil twilight = sparse dashes).

Includes 34 unit tests covering polar-coordinate math, timezone-aware
hour conversion, sun-time band resolution (astronomy path, weather
fallback, polar day, (0,0) sentinel), and end-to-end render across
several edge cases (no weather, no coords, all-day events, events
outside the dial). Pixel snapshot baseline regenerated.

### CODE REVIEW

File: src/render/components/almanac_panel.py [hunk 1]
```diff
@@ -0,0 +1,786 @@
...
+
+    # Upcoming birthdays inside the next two weeks
+    if data.birthdays:
+        for b in data.birthdays:
+            try:
+                bday_this_year = b.date.replace(year=today.year)
+            except ValueError:
+                continue
+            if bday_this_year < today:
+                bday_this_year = b.date.replace(year=today.year + 1)
```
[the remark this answers is not in the document]
Comment (gkoch02) [reply src/render/components/almanac_panel.py]: Good catch — both branches were broken. Fixed in 7171ef5: the no-leap-day-in-this-year branch now falls back to Feb 28 (matching the convention `birthday_bar.py` already uses) instead of dropping the entry, and the `year=today.year + 1` rollover is wrapped in the same try/except so it can't crash when going from a leap year into a non-leap one. Added three regression tests covering both directions.

---
_Generated by [Claude Code](https://claude.ai/code)_
